includes/collections/class-collection-taxonomy.php

```diff
@@ -214,4 +215,42 @@ public static function find_term_by_name( $term_name, $meta_conditions = [] ) {
 
 		return get_terms( $args );
 	}
+
+	/**
+	 * Add the Collections taxonomy to the custom taxonomies array.
+	 *
+	 * @param array $custom_taxonomies Array of custom taxonomies.
+	 * @return array Modified array of custom taxonomies.
+	 */
+	public static function add_collections_taxonomy_to_blocks( $custom_taxonomies ) {
+		$collections_taxonomy = [
+			'slug'  => self::TAXONOMY,
+			'label' => __( 'Collections', 'newspack-plugin' ),
+		];
+
+		// Don't add a duplicate if this taxonomy is already in the list.
+		foreach ( $custom_taxonomies as $tax ) {
+			if ( $tax['slug'] === $collections_taxonomy['slug'] ) {
+				return $custom_taxonomies;
+			}
+		}
+
+		// Capture where Collection Sections is in the sidebar.
+		$point_of_insertion = null;
+		foreach ( $custom_taxonomies as $index => $tax ) {
+			if ( isset( $tax['slug'] ) && $tax['slug'] === 'newspack_collection_section' ) {
+				$point_of_insertion = $index;
+				break;
+			}
+		}
```

Comment:
You can avoid double-looping the same array and merge the logic:

```suggestion
		$point_of_insertion = null;
		foreach ( $custom_taxonomies as $index => $tax ) {
			if ( isset( $tax['slug'] ) ) {
				if ( $tax['slug'] === $collections_taxonomy['slug'] ) {
					return $custom_taxonomies;
				}
				if ( $tax['slug'] === Collection_Section_Taxonomy::get_taxonomy() ) {
					$point_of_insertion = $index;
				}
			}
		}
```

Reply:
Ah that's much nicer - updated!